pkg/controller/installation/core_controller.go

```diff
@@ -1686,6 +1686,20 @@ func (r *ReconcileInstallation) setDefaultsOnFelixConfiguration(ctx context.Cont
 		updated = true
 	}
 
+	// MKE uses a vxlanVNI:4096 and vxlanPort:4789 for its docker swarm vxlan.
+	// This results in a conflict with calico's VXLAN and the vxlan.calico interface
+	// gets deleted. To fix this we change the vxlanVNI to 10000 as recommended by
+	// MKE docs (https://docs.mirantis.com/mke/3.7/cli-ref/mke-cli-install.html).
+	if install.Spec.KubernetesProvider == operator.ProviderDockerEE {
+		// Set bpfHostConntrackBypass to false for eBPF dataplane to work with MKE
+		if install.Spec.BPFEnabled() {
+			disableBPFHostConntrackBypass(fc)
+		}
+		vxlanVNI = 10000
+		fc.Spec.VXLANVNI = &vxlanVNI
```

Comment:
The code here should only be setting values if they are not already set.
That will also hopefully ensure that this change won't impact functionality in existing clusters.

Does `fc.Spec.VXLANVNI` get defaulted by CalicoNode/Felix? If not that could be an issue with impacting existing clusters, unless it is safe to change VXLANVNI on the fly.